### ISSUE TITLE
Qt apps do not add space after committing preedit

### ISSUE THREAD

**Issue by PedroArvela** [mike-fabian/ibus-typing-booster#12]
Using `ibus-typing-booster` 1.5.23 on Fedora 25 with Plasma 5 desktop and `ibus-qt` 1.3.3, all from the default repositories.

**Environment variables:**

```bash
env | grep ibus

QT_IM_MODULE=ibus
GTK_IM_MODULE=ibus
XMODIFIERS=@im=ibus
```

**Steps to reproduce:**

1. Open Qt-based application (i.e. kwrite)
2. Set Typing Booster as typing method
3. Write word which creates preedit suggestions
4. Press space

**Expected outcome:**

The selected candidate should be committed, followed by a space.

**Actual outcome:**

The selected candidate is committed, without a space following.

---

The same experiment in GTK based applications works as expected.

**Comment by mike-fabian** [comment]
This is Qt5, right? For Qt4 there is the workaround to use XIM:

export QT_IM_MODULE=xim
export XMODIFIERS=@im=ibus

But Qt5 does not support XIM anymore.

This problem is caused by the broken implementation of forward_key_event() in ibus-qt.

**Comment by mike-fabian** [comment]
I am thinking what kind of  workaround for Qt5 I  could do  until  this problem is fixed in ibus-qt...

**Comment by PedroArvela** [comment]
Exactly, this is when using Qt5 based apps.

**Comment by mike-fabian** [comment]
https://bugreports.qt.io/browse/QTBUG-58220

**Comment by mike-fabian** [comment]
There seems to be no way to find out from an ibus input method like ibus-typing-booster
whether the program currently typed into is Qt, Gtk, or using XIM (like Qt when using the
XIM module or X11 applications like xterm or emacs).

I can make it work for Qt with  a small change using “return False” instead of forward_key_event()
here:

https://github.com/mike-fabian/ibus-typing-booster/blob/master/engine/hunspell_table.py#L2806

That makes it work for Qt4 (when the ibus module is used) and Qt5 but breaks it
for Qt4 when the xim module is used and for all XIM using programs like xterm.

If I could test that at runtime from ibus-typing-booster, I could use either "return False"
or forward_key_event() there as appropriate. But unfortunately this is not possible.

So I am thinking about adding another option to the setup tool like this:

[✓] Enable workaround for Qt im module

And the tooltip for that option would explain that using  this option would break it for
XIM. I.e. the user could than at least decide whether he wants to have it working in 
Qt or in X11 programs.

What do you think?

I have no better idea at the moment ...

**Comment by PedroArvela** [comment]
I don't see a better approach either. I agree that the checkbox is the best option at the moment, at least as long as the bug isn't fixed from Qt's side.

**Comment by mike-fabian** [comment]
I have added this option in my working branch already:
https://github.com/mike-fabian/ibus-typing-booster/commit/9b115d2027891a2a9e192d4b010a0f1d065a2aed

Releasing this will take a bit more time because I need to fix breakage introduced to the UI
by a glade bug. Glade destroyed the user interface file when I saved it after adding
that option to the UI. I  opened a glade upstream bug for this:

https://bugzilla.gnome.org/show_bug.cgi?id=779806

In the long run I will probably stop using glade, but for now I’ll fix this with glade as quickly as I can
to be able to release this soon. Maybe tomorrow ...
